lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_ansible_method_base.rb

```diff
@@ -38,9 +38,11 @@ def running_in_state_machine?
       @workspace.root['ae_state_started'].present?
     end
 
-    def process_result
+    def process_result(task = nil)
       @aw.reload
       @workspace.update_workspace(@aw.output) if @aw.output
+      ansible_stats = task.task_results&.dig('ansible_stats')
+      @workspace.update_workspace('state_vars' => {'ansible_stats' => ansible_stats}) if ansible_stats
```

Comment:
@lfu the state vars typically store scalar data and not hashes.
This line might save us https://github.com/ManageIQ/manageiq-automation_engine/blob/b4f7c7a9931d9a70ba8c918be2c4e2c7436e259b/lib/miq_automation_engine/engine/miq_ae_engine.rb#L66 but you might want to test it out so that it can preserve the nested hashes